typespec-tests/tsp/arm/arm.tsp

```diff
@@ -0,0 +1,100 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+@armProviderNamespace
+@service({
+  title: "ArmResource",
+  version: "2023-11-01",
+})
+@doc("Arm Resource Provider management API.")
+@useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+@useDependency(Azure.Core.Versions.v1_0_Preview_1)
+namespace Cadl.ArmResource;
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ResourceManager;
+
+//----------------------- Child Resources -----------------------
+@doc("Subresource of Top Level Arm Resource.")
+@parentResource(TopLevelArmResource)
+model ChildResource is TrackedResource<ChildResourceProperties> {
+  @key("childResourceName")
+  @doc("ChildResources")
+  @visibility("read")
+  @path
+  @segment("childResources")
+  @pattern("^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$")
+  name: string;
+}
+
+@doc("Child Resource Properties.")
+model ChildResourceProperties {
+  @visibility("read")
+  @doc("Provisioning State of Top Level Arm Resource")
+  provisioningState?: ProvisioningState;
+}
+
+//----------------------- Top Level Arm Resource -----------------------
+@resource("topLevelArmResources")
+model TopLevelArmResource is TrackedResource<TopLevelArmResourceProperties> {
+  @key("topLevelArmResourceName")
+  @path
+  @segment("topLevelArmResources")
+  @doc("arm resource name for path")
+  @pattern("^[A-Za-z0-9]([A-Za-z0-9-_.]{0,62}[A-Za-z0-9])?$")
+  name: string;
+}
+
+@doc("Top Level Arm Resource Properties.")
+model TopLevelArmResourceProperties {
+  @doc("Configuration Endpoints.")
+  @visibility("read")
+  configurationEndpoints?: string[];
+
+  @visibility("read")
+  @doc("ChildResources References List")
+  childResources?: ResourceId[];
+
+  @visibility("read")
+  @doc("The status of the last operation.")
+  provisioningState?: ProvisioningState;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
+@lroStatus
+enum ProvisioningState {
+  Provisioning,
+  Updating,
+  Deleting,
+  Accepted,
+  Succeeded,
+  Failed,
+  Canceled,
+}
+
+@doc("Resource ID definition used by parent to reference child resources.")
+model ResourceId {
+  @doc("Resource ID of child resource.")
+  id: string;
+}
```

Comment:
SubResource?

Or https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-resource-manager/lib/models.tsp#L108 ?

Basically I didn't see a reason this kind of basic model need to be defined...

Reply:
Maybe check with Chenjie about whether there is `SubResource` in resourcemanager lib.

Reply:
Chenjie is aware of ResourceIdentifier:
https://github.com/Azure/autorest/blob/faf5c1168232ba8a1e8fe02fbc28667c00db8c96/packages/extensions/openapi-to-typespec/test/arm-networkanalytics/tsp-output/models.tsp#L226

I found Fleet has such use case:
https://github.com/Azure/azure-rest-api-specs/blob/f776434f63fb6505926273db8d4f9a93b75ee4a1/specification/containerservice/Fleet.Management/update/run.tsp#L42-L47

swagger equivalent:
https://github.com/Azure/azure-rest-api-specs/blob/f776434f63fb6505926273db8d4f9a93b75ee4a1/specification/containerservice/resource-manager/Microsoft.ContainerService/fleet/preview/2023-08-15-preview/fleets.json#L2182

.Net treat it as ResourceIdentifier:
https://github.com/Azure/azure-sdk-for-net/blob/b9469adaa8f5cb0076ffcb9b71c667519a300b1d/sdk/fleet/Azure.ResourceManager.ContainerServiceFleet/src/Generated/ContainerServiceFleetUpdateRunData.cs#L83

Not SubResource(in VM):
https://github.com/Azure/azure-sdk-for-net/blob/b9469adaa8f5cb0076ffcb9b71c667519a300b1d/sdk/compute/Azure.ResourceManager.Compute/src/Generated/VirtualMachineData.cs#L163

Though he's not sure how to represent `SubResource` in tsp either. I'll remove the reference property here till we reach agreement on how to define it.